### ISSUE TITLE
Run as eclipse plugin.

### ISSUE THREAD

**Issue by lombokissues** [rzwitserloot/lombok#397]
_Migrated from Google Code (issue 324)_

**Comment by Katharsas** [comment]
This would also make make it so that Lombok can automatically be added to an Eclipse Installation by Eclipse Installer (Oomph) configuration i think (and potentially synced to all Eclipse installations of a user).

**Comment by bmuskalla** [comment]
Adding the vm arguments during plugin installation should be possible using p2 touchpoints in the p2.inf, see https://help.eclipse.org/neon/topic/org.eclipse.platform.doc.isv/guide/p2_actions_touchpoints.html?cp=2_0_20_1_3 and 
Seems someone got this to work already: https://github.com/adangel/custom-eclipse/blob/cb895947e620f3b502fd0c27a790071489b64f72/lombok/META-INF/p2.inf

**Comment by jordiboehmelopez** [comment]
In our organisation we automatically distribute Eclipse using Oomph.
The custom installer provided by Lombok does not quite fit into the workflow.
To solve this I created a p2 repository.
This enables us, and every other eclipse user out there, to install the Lombok Agent into eclipse using standard installer mechanisms.
I created a pull request here: #2371
Here is the p2 repository which was created with the sources of the pull request:
https://jordiboehmelopez.github.io/lombok-eclipse/
